src/FirebaseArduino.h

```diff
@@ -24,18 +24,97 @@
 #define FIREBASE_JSONBUFFER_SIZE 200
 #endif // FIREBASE_JSONBUFFER_SIZE
 
+/**
+ * Gateway class for Arduino clients to interact with the Firebase backend.
+ * This implementation is designed to follow Arduino best practices and favor
+ * simplicity over all else. 
+ * For more complicated usecases and more control see the Firebase class in 
+ * Firebase.h.
+ */
 class FirebaseArduino {
  public:
+  /**
+   * Must be called first. This sets the class up for use.
+   * \param host Your firebase db domain name, usually X.firebaseIO.com.
+   * \param auth Optional authentication for the db, a secret or token.
+   */
   void begin(const char* host, const char* auth = "");
+
+  /**
+   * Adds the data in value to the list located at path. Equivilent to the
+   * REST API's POST.
+   * You should check success() after calling.
+   * \param path The path inside of your db to the list you wish to update.
+   * \param value Data that you wish to add to the list.
+   * \return The child name where the data was written.
+   */
   String push(const String& path, const JsonVariant& value);
+
+  /**
+   * Writes the data in value to the node located at path Equivilent to the
+   * REST API's PUT.
+   * You should check success() after calling.
+   * \param path The path inside of your db to the node you wish to update.
+   * \param value Data that you wish to write.
+   */
   void set(const String& path, const JsonVariant& value);
+
+
+  /**
+   * Gets the value located at path.
+   * You should check success() after calling.
+   * \param path The path inside of your db of the node you wish to retrieve.
+   * \return The data located at that path. This may either be a single element
+   * or it may be a json structure. Will only be populated if success() is true.
+   */
   FirebaseObject get(const char* path);
+
+  /**
+   * Remove the variable, and possibly entire tree, located at path.
```

Comment:
s/variable/node/ or object.

Or maybe just reuse the wording from the docs:
https://www.firebase.com/docs/web/api/firebase/remove.html

> Removes the data at this Firebase location.